man/rn2ncid.conf.5
.\" %W% %G%
.TH RN2NCID.CONF 5
.SH NAME
.B rn2ncid.conf
- rn2ncid configuration file
.SH DESCRIPTION
The \fIrn2ncid.conf\fR file contains the configuration information for
\fIrn2ncid\fR.
.PP
The \fIrn2ncid.conf\fR file understands 3 types of lines:
.TP
.B blank line
ignored
.TP
.B comment line, beginning with '#'
ignored
.TP
.B VARIABLE = VALUE
VALUE can contain spaces if it is between quotes.
.PP
See \fIrn2ncid.conf\fR for all default values.
.SH VARIABLES
.TP
.B verbose = \fIlevel\fR
Level is in the range 1-9. Not all verbose levels are used.
.TP
.B ncidaddr = \fIhost address\fR
The NCID server host address can be an IP address or a resolvable host name.
.TP
.B ncidport = \fIport\fR
The NCID server port number.
.TP
.B delay = \fIseconds\fR
If the connection to the NCID server is lost, this specifies the number
of seconds between reconnection attempts.
.TP
.B cellport = \fIport\fR
The listen port as specified in the Remote Notifier application.
.TP
.B reject = \fIfrom address[,from address][,from address]...\fR
SMS and MMS messages usually have a 'from' address and depending on 
the carrier this could be a number or an email address. Specify a comma
separated list of addresses that will NOT be transmitted by the NCID 
server to all listening NCID clients. This is used to prevent endless 
loops which could result in excessively high data or text charges
by your cell phone carrier.
.PP
Remote Notifier sends the 'from' address in its 'data' (fifth) field.
See NOTES below.
.PP
An example of an endless loop would be:
.RS 5
.IP - 3
cell phone gets SMS
.IP - 3
Remote Notifier sends out the SMS
.IP - 3
rn2ncid gets the SMS and sends it to the NCID server as 'MSG'
.IP - 3
the ncid-page module gets this 'MSG' and forwards it back to the cell phone
.IP - 3
cell phone gets the 'MSG' as an SMS
.IP - 3
cycle repeats forever
.RE
.PP
To prevent this endless loop, suppose you have ncid-page.conf
configured to send SMS messages to your cell 'from' your email
address of: johndoe@yahoo.com
.PP
.PP
You might want to set 'reject' to be:
.PP
.PP
.RS 3
reject = johndoe@yahoo.com
.RE
.SH NOTES
Remote Notifier sends six pieces of information:
.RS 5
.IP - 3
(1) version (e.g., 'v2')
.IP - 3
(2) unique device id in hex
.IP - 3
(3) unique notification id in hex
.IP - 3
(4) event type (BATTERY, PING, SMS, etc.)
.IP - 3
(5) data (terse information)
.IP - 3
(6) event contents (verbose free-form text, can be multi-line)
.RE
.SH EXAMPLES
Set the NCID server IP address:
.PP
.RS 3
ncidaddr = 192.168.1.20
.RE
.PP
Set the Remote Notifier listen port
.PP
.RS 3
cellport = 10600
.RE
.PP
Reject SMS/MMS messages to manage text blocking
.PP
.RS 3
reject = 9999
.RE
.SH SEE ALSO
rn2ncid.1


doc/CHANGES
ncid - Network CallerID

Version 0.87
    - README: updated
    - Makefile: added test: $(DOCDIR) does not exist, create it
    - All GPL license headers in files were updated to GPL 3
    - client/ncid.sh: logClock{} changed {^(CID|HUP|OUT)} to {^(?:CID|HUP|OUT)}
                      added "Line Labels" in help menu and labInfo {}
                      checkTypes() added new types and changed type numbers
                      getCID(): added new types and changed if statements
                                for new Type numbers
    - debian/Makefile: updated
    - debian/README.Debian: updated
    - debian/*.init: removed "S" from default STOP
    - debian/wc2ncid.init: removed configuration stuff, changed sbin to sbin
    - debian/yac2ncid.init: removed configuration stuff, changed sbin to sbin
    - debian/rn2ncid.init: new
    - doc/INSTALL-Ubuntu: updated
    - doc/INSTALL-Raspbian: updated
    - doc/LICENSE: updated to GPL 3
    - doc/Setup-rn2ncid: new
    - doc/Verbose-rn2ncid: new
    - Fedora/Makefile: updated
    - Fedora/README.Fedora: updated
    - Fedora/rn2ncid.init: new
    - Fedora/rn2ncid.service: new
    - FreeBSD/Makefile: updated
    - FreeBSD/README.FreeBSD: updated
    - FreeBSD/rn2ncid.rc: new
    - gateway/Makefile: added $(EXTRA_CFLAGS) to CFLAGS definition
    - gateway/README.Gateways: renamed to README.gateways and updated
    - gateway/*.[ch]: updated license in headers
    - gateway/*.pl: updated license in comments
    - gateway/sip2ncid.h: added UNAUTHORIZED define
    - gateway/sip2ncid.c: processPackets() added UNAUTHORIZED section
    - gateway/wc2ncid.conf: - improved comment wording
                            - changed nciddelay to delay
    - gateway/wc2ncid.pl: - moved test mode just after Command Line Processing
                          - moved logfile and config file messages to
                            after start message
                          - added cleanup(), called by errorExit() & sigHandle()
                          - added delay option to pod documentation wording,
                            improved some pod wording
                          - changed $nciddelay to $delay, added $Delay,
                            added --delay|-d option
    - gateway/rn2ncid.pl: new
    - gateway/rn2ncid.config: new
    - Mac/Makefile: updated
    - Mac/README.Mac: updated
    - Mac/net.sourceforge.rn2ncid.plist: new
    - man/Makefile: updated
    - man/ncid-*.1: updated
    - man/ncid.1: updated
    - man/ncid.conf.5: updated
    - man/ncidd.8: updated
    - man/yac2ncid.1: updated
    - man/rn2ncid.conf.5: new
    - modules/ncid-*.sh: updated
    - modules/ncid-*.conf: updated
    - modules/ncid-notify.sh: changed line one from /bin/sh to /bin/bash
                              minor changes to verbose output and comments
    - modules/ncid-speak.sh: added "AreaCodeLength" code to speak the area
                             code instead or "NO NAME" or entire number
    - modules/ncid-speak.conf-in: added "AreaCodeLength" and comments
    - server/ncidd.c: writeLog() - moved "logMsg(LEVEL3, msgbuf);" to start
                      formatCID() - moved cidnoname test inside
                                    if (!(cid.status & CIDNMBR))
                                 - added PID:
                      added BLKLINE, NOTLINE, PIDLINE to *lineTags[]
                      doPoll: - added code for a NOT: message
    - server/ncidd.h: added defines for PID, PIDLINE, NOTLINE, CALLPID
    - test/README.test: updated
    - test/*: renamed test programs and test data
    - test/test-rn2ncid: new
    - test/test-rn2ncid.data: new
    - TiVo/ncid-tivo.sh:  if no line label, display $CIDNAME/$CIDNMBR
                          else display "$CIDNAME $CIDNMBR/$CIDTYPE $CIDLINE"

Version 0.86.1
    - debian/changelog: updated
    - debian/control: for package ncid-client added "curl" to Depends

Version 0.86
    - README: updated
              added redhat: and redhat-install:
    - attic/*: new
    - client/ncid.sh: added more doVerbose statements and levels
                      changed 300* doVerbose message to any one of four
                      added line count to "End of call log" message
                      removed the --raw | -R option
                      fixed and improved call on a ring number
                      fonts used depend on platform and OS
                      allow resizing of window
                      allow font size changes
                      time format in history window can be either 12 and 24
                      new menu called Preferences for Fonts and time format
                      menu item to save window size
                      menu item to save window size and location
                      added the "-h | --help" option
                      added the "-v | --version" option
                      added the "-w | --wakeup" option
                      getCID{}: added calls to ncid-wakeup module
                      changed "-V" to be the "-v | --verbose" option
                      makeWindow{}: improved widget and menu code
                      added saveSize{} write_rc_file{} logFont{} changeFont{}
                      getField{}: changed regex to include "*" and "-"
    - client/ncid.conf-in: removed MsgFlag variable and comments
                           removed Raw variable and comments
                           added comments to CallOnRing section
                           added comments to Ring Options section
                           new "WakeUp" section
                           new "WrapLines" section
    - debian/Makefile: updated
    - debian/wc2ncid.init: new
    - debian/ncidsip.init: removed
    - doc/CONTRIBUTORS: updated
    - doc/NCID-SDK.odt: updated
    - doc/Setup-ncidsip: removed
    - doc/Setup-sip2ncid: updated
    - doc/Setup-wc2ncid: new
    - doc/INSTALL-Redhat: new
    - doc/INSTALL*: updated
    - doc/INSTALL-RPi: new
    - Fedora/Makefile: updated

gateway/rn2ncid.pl
#!/usr/bin/perl

# rn2ncid - Android smart phone app 'Remote Notifier' to NCID gateway

# Copyright (c) 2005-2013
#  by John L. Chmielewski <jlc@users.sourceforge.net>
#     Todd Andrews <tandrews@users.sourceforge.net>

# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.

# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

use strict;
use warnings;

use POSIX qw(strftime);
use Getopt::Long qw(:config no_ignore_case_always);
use File::Basename;
use Config::Simple;
use Pod::Usage;
use IO::Socket::INET;
use IO::Select;

my $prog = basename($0);
my $confile = basename($0, '.pl');
my $VERSION = "(NCID) XxXxX";

my $ConfigDir = "/usr/local/etc/ncid";
my $ConfigFile = "$ConfigDir/$confile.conf";

# Constants
my $CALLTYPE = "PID";

my ($ncidaddr, $ncidport) = ('localhost', 3333);
my ($peerport, $peeraddr);
my $ncidhost = "";
my $ncidsock = undef;
my $ncidline = undef;
my $cellport = 10600;
my $cellPort = undef;
my $cellsock = undef;
my $celldata;
my $defaultTO = 0.2;
my $timeout = $defaultTO;
my $selectTO = undef;
my $delay = 30;
my $Delay = undef;
my $logfile = basename($0, '.pl');
   $logfile = "/var/log/" . $logfile . ".log";
my ($logfileMode, $logfileModeEnglish);
my $logfileAppend;
my $logfileOverwrite;
my $debug;
my $verbose = 1;
my $Verbose = undef;
my ($help, $man, $version);
my $pidfile = "";
my $savepid;
my $pid;
my $test;
my $fileopen;
my $select;
my @ready;
my $rh;
my ($pos, $num, $cnt, $loc, $gotnum);
my $match = undef;
my $addr;
my $cfg;
my $linenum = 1;
my $ignoreTO;
my %config;
my @reject;

my $date = strftime("%m/%d/%Y %H:%M:%S", localtime);

# command line processing
my @save_argv = @ARGV;
Getopt::Long::Configure ("bundling");
my ($result) = GetOptions("ncidhost|n=s" => \$ncidhost,
               "configfile|C=s" => \$ConfigFile,
               "logfile-append|l=s" => \$logfileAppend,
               "logfile-overwrite|L=s" => \$logfileOverwrite,
               "debug|D" => \$debug,
               "delay|d" => \$Delay,
               "help|h" => \$help,
               "man|m" => \$man,
               "verbose|v=i" => \$Verbose,
               "version|V" => \$version,
               "test|t" => \$test,
               "cellport|c=s" => \$cellPort,
               "pidfile|p=s" => \$pidfile
             ) || pod2usage(2);
die "$prog $VERSION\n" if $version;
pod2usage(-verbose => 1, -exitval => 0) if $help;
pod2usage(-verbose => 2, -exitval => 0) if $man;

# reading configuration file after command line processing
# is necessary because the command line can change the
# location of the configuration file
$cfg = new Config::Simple($ConfigFile);
if (defined $cfg) {
    # opened config file
    %config = $cfg->vars();
    $verbose = $config{'default.verbose'};
    $ncidaddr = $config{'default.ncidaddr'};
    $ncidport = $config{'default.ncidport'};
    $delay = $config{'default.delay'};
    $cellport = $cfg->param('default.cellport');
    @reject = $cfg->param('default.reject');
}

if ($test) {
    $debug = 1;
    $verbose = 3;
}

# these command line values override the configuration file values
$ncidport = $1 if $ncidhost =~ s/:(\d+)//;
$ncidaddr = $ncidhost if $ncidhost;
$cellport = $cellPort if $cellPort;
$delay    = $Delay if $Delay;
$verbose = $Verbose if defined $Verbose;

$logfileMode = ">>"; # default to append
$logfileModeEnglish = "Appending to";

if ($logfileAppend and $logfileOverwrite) { $logfileOverwrite = undef; }

if ($logfileOverwrite) {
   $logfileMode = ">";
   $logfileModeEnglish = "Overwriting";
   $logfile = $logfileOverwrite;
} else {
  if ($logfileAppend) {
     $logfile = $logfileAppend;
  }
}
  
if (open(LOGFILE, "$logfileMode$logfile")) {
    LOGFILE->autoflush(1); # make LOGFILE handle 'hot', i.e., no buffering
    $fileopen = 1;
}

logMsg(1, "Started: $date\n");

# log command line and any options on separate lines
my $cl = "Command line: " . $0;
for my $arg (@save_argv) {
    if ( '-' eq substr($arg, 0, 1)) {
        logMsg(1, "$cl\n");
        $cl = "              $arg";
    } else {
        $cl = $cl . " " . $arg;
    }
}
logMsg(1, "$cl\n");

if ($fileopen) {logMsg(1, "Logfile: $logfileModeEnglish $logfile\n");}
else {logMsg(1, "Could not open logfile: $logfile\n");}

if (defined $cfg) {logMsg(1, "Processed config file: $ConfigFile\n");}
else {logMsg(1, "Config file not found: $ConfigFile\n");}

logMsg(1, "Gateway: $prog version $VERSION\n");
logMsg(1, "Verbose level: $verbose\n");
logMsg(1, "Debug mode\n") if ($debug);

if ($test) {logMsg(1, "Test mode\nNot sending data to NCID\n");}

&doPID;

if (@reject) {logMsg(1, "Ignoring messages from: @reject\n");}
else {logMsg(1, "No messages rejected.\n");}

$SIG{'HUP'}  = 'sigHandle';
$SIG{'INT'}  = 'sigHandle';
$SIG{'QUIT'} = 'sigHandle';
$SIG{'TERM'} = 'sigHandle';
$SIG{'PIPE'} = 'sigIgnore';

$select = IO::Select->new();

# $select undefined if could not create new object
errorExit("ERROR in Select Object Creation : $!") if !defined $select;

&connectNCID if ! $test;

&connectPhone;

# get a set of readable handles, block until at least one is ready
while (1) {
    if (!(@ready = $select->can_read($selectTO))) {
        # select timeout
        connectNCID();
        if (defined  $ncidsock) {
            $selectTO = undef;
            logMsg(1, "Listening at port $cellport\n");
        }
    }
    foreach $rh (@ready) {
        if (defined $ncidsock && $rh == $ncidsock) {
          # NCID server Caller ID
          $ncidline = <$rh>;
          if (!defined $ncidline) {
            $select->remove($ncidsock);
            $selectTO = $delay;
            logMsg(1, "NCID server at $ncidaddr:$ncidport disconnected\n");
            logMsg(1, "Trying to reconnect every $delay seconds\n");
          }
          else {logMsg(5, $ncidline);}
        }
        elsif ($rh == $cellsock) {
          # Smart phone CID or message
          my $datasock = $cellsock->accept();
          my $ret = $datasock->recv($celldata, 1024);
          close($datasock);
          logMsg(3, "$celldata\n");
          &doLine;
        }
    }
}

sub connectNCID {
  $ncidsock = IO::Socket::INET->new (
    Proto    => 'tcp',
    PeerAddr => $ncidaddr,
    PeerPort => $ncidport,
  );

  # $ncidsock undefined if could not connect to server
  if (!defined $ncidsock) {
    if (defined $selectTO) {return;}
    else {errorExit("NCID server: $ncidaddr:$ncidport $!");}
  }

  logMsg(1, "Connected to NCID server at $ncidaddr:$ncidport\n");
  my $greeting = <$ncidsock>;
  logMsg(1, "$greeting");

  # read and discard cidcall log sent from server
  while (<$ncidsock>)
  {
    # a log file may or nay not be sent
    # but a 300 message is always sent
    last if /^300/;

    logMsg(5, $_);
  };
  logMsg(1, $_); # 300 message

  $select->add($ncidsock);
}

sub connectPhone {
  $cellsock = IO::Socket::INET->new (
    Proto     => 'tcp',
    Listen    => 5,
    LocalPort => $cellport,
    Reuse     => 1
  ) or errorExit("Could not listen at port: $cellport $!");

  logMsg(1, "Listening at port $cellport\n");

  $select->add($cellsock);
}

sub doLine {
    my $nciddate = strftime("%m%d%H%M", localtime);
    my ($type, $f5, $f6, $msg, $ncidname, $phoneid, $ncidnmbr, $logmsg);

    # Known message types: RING PING BATTERY SMS MMS VOICEMAIL
    ($phoneid, $type, $f5, $f6) = $celldata =~
        /\/\w+(\w\w\w\w)\/\w+\/(\w+)\/(.*?)\/(.*)/;

    $type="NULL" if !defined $type;
    logMsg(3, "Detected type: $type\n");
    
    if ($type =~ /RING/) {
        # incoming call
        ($ncidname) = $f6 =~ /(.*?)(,\s|\s-)/;
        $logmsg = $msg =
            sprintf("CALL: ###DATE%s...CALL%s...LINE%s...NMBR%s...NAME%s+++",
                    $nciddate, $CALLTYPE, $phoneid, $f5, $ncidname);
    } elsif ($type =~ /PING|BATTERY/) {
        # internal message
        $logmsg = $msg = sprintf("NOT: PHONE %s: %s %s", $phoneid, $type, $f6);
    } elsif ($type =~ /SMS|MMS/) {
        # incoming message
        $logmsg = $msg = sprintf("NOT: PHONE %s: %s from %s",
            $phoneid, $type, $f5);
    } elsif ($type =~ /VOICEMAIL/) {
        # voice message
        $logmsg = $msg = sprintf("NOT: PHONE %s: %s", $phoneid, $type);
    } else {
        # unknown message
        $logmsg = "$type message not sent to NCID";
    }

    logMsg(3, "$logmsg\n");

    if ($msg) {
        my $norej = 1;
        foreach my $rej (@reject) {
            if($rej eq $f5) {
                $norej = 0;
                $logmsg = "rejected $rej";
                logMsg(3, "$logmsg\n");
            }
        }
        if ($norej && !$test && !defined $selectTO) {
            print $ncidsock $msg, "\r\n";
        }
    }
}

sub doPID {
    # Only create a PID file if $pidfile contains a file name
    if ($pidfile ne "") {
        if (-e $pidfile) {
            # only one instance per computer permitted
            unless (open(PIDFILE, $pidfile)) {
                errorExit("pidfile exists and is unreadable: $pidfile\n");
            }
            $savepid = <PIDFILE>;
            close(PIDFILE);
            chop $savepid;

            # Check PID file to see if active PID in it
            # Does not work for Windows
            if (-d "/proc") {
                if (-d "/proc/$savepid") {
                    errorExit("Process ($savepid) already running: $pidfile\n");
                } else {
                    logMsg(1, "Found stale pidfile: $pidfile\n");
                }
            } else {
                my $ret = `ps $savepid 2>&1`;
                if ($? == 0) {
                    errorExit("Process ($savepid) already running: $pidfile\n");
                } elsif ($? != -1) {
                    logMsg(1, "Found stale pidfile: $pidfile\n");
                } else {
                    logMsg(1, "ps command not found\n");
                }
            }
        }

        if (open(PIDFILE, ">$pidfile")) {
            print(PIDFILE "$$\n");
            $pid = $$;
            close(PIDFILE);
            logMsg(1, "Wrote pid $pid in $pidfile\n");
        } else { logMsg(1, "Could not write pidfile: $pidfile\n"); }
    }   
    else {logMsg(1, "Not using PID file\n");}
}

sub logMsg {
    my($level, $message) = @_;

    if (!defined $message) {print "Oops, unexpected exit\n"; exit 1}

    # write to STDOUT
    print $message if $debug && $verbose >= $level;

    # write to logfile
    print LOGFILE $message if $fileopen && $verbose >= $level;
}

sub cleanup() {
    close($ncidsock) if $ncidsock;
    close($cellsock) if $cellsock;
    if ($pid) {
        unlink($pidfile);
        logMsg(1, "Removed $pidfile\n");
    }
}

sub sigHandle {
    my $sig = shift;
    cleanup();
    my $date = strftime("%m/%d/%Y %H:%M:%S", localtime);
    logMsg(1, "\nTerminated $date: Caught SIG$sig\n");
    close(LOGFILE);
    exit(0);
}

sub sigIgnore {
    my $sig = shift;
    my $date = strftime("%m/%d/%Y %H:%M:%S", localtime);
    logMsg(1, ": Ignored SIG$sig: $date\n");
}

sub errorExit {
    logMsg(1, "@_");
    cleanup();
    my $date = strftime("%m/%d/%Y %H:%M:%S", localtime);
    logMsg(1, "\nTerminated: $date\n");
    close(LOGFILE);
    exit(-1);
}

=head1 NAME

rn2ncid - Android smart phone app 'Remote Notifier' to NCID gateway

=head1 SYNOPSIS

rn2ncid [--debug|-D]
        [--delay|-d <seconds>]
        [--help|-h]
        [--logfile-append|-l <filename>]
        [--logfile-overwrite|-L <filename>]
        [--configfile|-C <filename>]
        [--man|-m]
        [--ncidhost|-n <[host][:port]>]
        [--test| -t]
        [--pidfile|-p <filename>]
        [--verbose|-v <1-9>]
        [--version|-V]
        [--cellport|-c <port>]

=head1 DESCRIPTION

The B<rn2ncid> gateway obtains Caller ID and messages from a cell
phone.  It uses an Android app called B<Remote Notifier for Android>
to obtain the information and send it to the NCID server.  The server
then sends the CID information to the NCID clients.

The B<Remote Notifier for Android> app uses a 16 digit hex number to
identify the smart phone.  The B<rn2ncid> gateway uses 4 of the least
significant digits as the phone id.  Therefore you can run the app
in multiple smart phones without needing to configure them.

The phone id can be aliased by the NCID server so you can give each
phone a meaningful identification such a B<CELL>, or B<SP-1>, or
wharever.

The B<rn2ncid> configuration file is B</etc/ncid/rn2ncid.conf>.
See the rn2ncid.conf man page for more details.  If you are
also using B<ncid-page> or B<ncid-notify>.  you need to configure
the B<reject> variable.

The B<rn2ncid> gateway can run on any computer, but normally it is run
on same box as the NCID server.  If it is not run on the same box as the
NCID server, you must configure the server IP address in the configuration
file.

=head2 OPTIONS

=over 2

=item -n <[host][:port]>, --ncidhost=<[host][:port]>

Specifies the NCID server.
Port may be specified by suffixing the hostname with :<port>.

Input must be <host> or <host:port>, or <:port>

Default:  localhost:3333

=item -c <port>, --cellport <port>

Specifies the port to listen on for messages from a smart phone.

Default 10600

=item -d <seconds>, --delay <seconds>

If the connection to the NCID server is lost,
try every <delay> seconds to reconnect.

Default: 30

=item -D, --debug

Debug mode, displays all messages that go into the log file.
Use this option to run interactively.

=item -h, --help

Displays the help message and exits.

=item -m, --man

Displays the manual page and exits.

=item -C, --configfile=<filename>

Specifies the configuration file to use.  The program will still run if
a configuration file is not found.

Default: /usr/local/etc/ncid/rc2ncid.conf

=item -l, --logfile-append=<filename>

=item -L, --logfile-overwrite=<filename>

Specifies the logfile name to write.  The program will still run if
it does not have permission to write to it.

If both options are present, --logfile-append takes precedence.

Default: Append to /var/log/rn2ncid.log

=item -p, --pidfile=<filename>

Specifies the pidfile name to write. The program will still run if
it does not have permission to write a pidfile. The pid filename that
should be used is /var/run/rc2ncid.pid.

Default: no pidfile

=item -t, --test

Test mode is a connection to the Whozz Calling Network Device
without a connection to the NCID server.  It sets debug mode
and verbose = 3.  The verbose level can be changed on the command line.

Default: no test mode

=item -v, --verbose <1-9>

Output information, used for the logfile and the debug option.  Set
the level to a higher number for more information.  Levels range from
1 to 9, but not all levels are used.

Default: verbose = 1

=item -V, --version

Displays the version.

=back

=head1 EXAMPLES

=over 4

=item Start lcdncid in test mode at verbose level 3

rn2ncid --test

=item Start rc2ncid in debug mode at verbose level 1

rn2ncid -D

=back

=head1 REQUIREMENTS

=over

=item https://play.google.com/store/apps/details?id=org.damazio.notifier&hl=en

The "Remote Notifier for Android" app on your Android device.

=back

perl 5.6 or higher,
perl(Config::Simple)

=head1 FILES

/etc/ncid/rn2ncid.conf

=head1 SEE ALSO

ncidd.8,
wc2ncid.1,
sip2ncid.8

=cut
